packages/core/src/js/feedback/FeedbackForm.types.ts

```diff
@@ -0,0 +1,143 @@
+import type { TextStyle, ViewStyle } from 'react-native';
+
+export interface FeedbackFormProps extends FeedbackGeneralConfiguration, FeedbackTextConfiguration, FeedbackCallbacks {
+  styles?: FeedbackFormStyles;
+}
+
+/**
+ * General feedback configuration
+ */
+export interface FeedbackGeneralConfiguration {
+  /**
+   * Should the email field be required?
+   */
+  isEmailRequired?: boolean;
+
+  /**
+   * Should the name field be required?
+   */
+  isNameRequired?: boolean;
+
+  /**
+   * Should the email input field be visible? Note: email will still be collected if set via `Sentry.setUser()`
+   */
+  showEmail?: boolean;
+
+  /**
+   * Should the name input field be visible? Note: name will still be collected if set via `Sentry.setUser()`
+   */
+  showName?: boolean;
+
+  /**
+   * Fill in email/name input fields with Sentry user context if it exists.
+   * The value of the email/name keys represent the properties of your user context.
+   */
+  useSentryUser?: {
+    email: string;
+    name: string;
+  };
+}
+
+/**
+ * All of the different text labels that can be customized
+ */
+export interface FeedbackTextConfiguration {
+  /**
+   * The label for the Feedback form cancel button that closes dialog
+   */
+  cancelButtonLabel?: string;
+
+  /**
+   * The label for the Feedback form submit button that sends feedback
+   */
+  submitButtonLabel?: string;
+
+  /**
+   * The title of the Feedback form
+   */
+  formTitle?: string;
+
+  /**
+   * Label for the email input
+   */
+  emailLabel?: string;
+
+  /**
+   * Placeholder text for Feedback email input
+   */
+  emailPlaceholder?: string;
+
+  /**
+   * Label for the message input
+   */
+  messageLabel?: string;
+
+  /**
+   * Placeholder text for Feedback message input
+   */
+  messagePlaceholder?: string;
+
+  /**
+   * Label for the name input
+   */
+  nameLabel?: string;
+
+  /**
+   * Message after feedback was sent successfully
+   */
+  successMessageText?: string;
+
+  /**
+   * Placeholder text for Feedback name input
+   */
+  namePlaceholder?: string;
+
+  /**
+   * Text which indicates that a field is required
+   */
+  isRequiredLabel?: string;
+
+  /**
+   * The title of the error dialog
+   */
+  errorTitle?: string;
+
+  /**
+   * The error message when the form is invalid
+   */
+  formError?: string;
+
+  /**
+   * The error message when the email is invalid
+   */
+  emailError?: string;
+}
+
+/**
+ * The public callbacks available for the feedback integration
+ */
+export interface FeedbackCallbacks {
+  /**
+   * Callback when form is closed and not submitted
+   */
+  onFormClose?: () => void;
```

Comment:
We should implement also the following, to provide the minimal subset of JS callbacks.

```js
  onSubmitSuccess: (data: FeedbackFormData) => {},
  onSubmitError: (error: Error) => {},
```

Now, onFormClose is also execute for successful submission.

Reply:
Good point @krystofwoldrich 👍 

> We should implement also the following, to provide the minimal subset of JS callbacks.

Added the missing methods with c295df6bf8ee926dcfd46ba7bbcf1988a0c40aff in https://github.com/getsentry/sentry-react-native/pull/4364 that implements some basic error handling (note that I plan to update this PR as described in its description with the next JS sdk).

> Now, onFormClose is also execute for successful submission.

Removed with 12b8ede2fd455367be7d3149095de77270b252b7 on the same PR.

Reply:
Perfect, thank you.